src/pages/HomePage.js

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate, useLocation } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { generate } from 'randomized-string';
+import { FaSearchLocation } from 'react-icons/fa';
+import fetchData from '../redux/covid/Api';
+import { fetchStats } from '../redux/covid/reducer';
+
+const HomePage = () => {
+  const countryStore = useSelector((store) => store.details);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (countryStore.length === 0) {
+      fetchData().then((response) => dispatch(fetchStats(response)));
+    }
+  }, [countryStore.length, dispatch]);
+
+  let continCovid = countryStore.filter((item) => item.continent === 'Europe');
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const search = query.get('search') || '';
+  continCovid = continCovid.filter((country) => country.country.includes(search));
+  const navigate = useNavigate();
+  const [searchValue, setSearchValue] = useState(search);
+
+  const countryFilterOnChange = (event) => {
+    navigate(event.target.value ? `?search=${event.target.value}` : '');
+    setSearchValue(event.target.value);
+  };
```

Comment:
- [ ] The search is not accurate due to the letter case. you may handle that by converting all county names and search values to lowercase.

With lowercase `i`
<img width="355" alt="Screen Shot 2022-09-03 at 12 22 16 PM" src="https://user-images.githubusercontent.com/40503704/188268461-4efc83c4-677f-43ac-bc64-45d476fad22e.png">

With uppercase `I`
<img width="355" alt="Screen Shot 2022-09-03 at 12 22 30 PM" src="https://user-images.githubusercontent.com/40503704/188268490-b7f6a39d-a626-4816-8c27-e38c33cb5b6f.png">

Reply:
I have included the unit test as requested by the code reviewer, but I have not been able to fix the search alphabet case issue as requested. Although isn't part of the requirement, I will be very glad if you can assist me on steps to fix it.

Reply:
Hi @Roseokpe, you can use this code for example:

```js
continCovid = continCovid.filter((country) => country.country.toLowerCase().includes(search.toLowerCase));
```
Hope it helps.🤝